### PR TITLE
[receiver/datadog]: Fix #23150 (missing span.Resource)

### LINKED ISSUE #23150: [receiver/datadog] span.Resource is dropped
### Component(s)

receiver/datadog

## Description

As of [pr 21764](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21764) the datadog receiver drops the span.Resource field - since this is where dd-java-agent put the db.statement field, this means that db statements are no longer available

## Steps to Reproduce

Instrument an application with dd apm, and check any db span

## Expected Result

span.Resource should be available as a tag on spans

## Actual Result

span.Resource is not available as a tag on spans

### Collector version

v0.78.0

## Environment
N/A

### OpenTelemetry Collector configuration

```yaml
N/A
```

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hughsimpson**
**Description:**
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23150 by include the original `dd.span.Resource` in the attributes of the translated span

**Link to tracking Issue:** [23150](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23150)

**Testing:** Added test that span.Resource is included in translate attributes

**Documentation:** Added note on `dd.span.Resource` to datadog receiver README

### CODE REVIEW

**Review by stevetursi — APPROVED** [review]

**Comment by hughsimpson** [comment]
Just dropping a comment so that the stale bot doesn't come over and close this pr this week

File: receiver/datadogreceiver/translator.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ func toTraces(payload *pb.TracerPayload, req *http.Request) ptrace.Traces {
 			newSpan.SetParentSpanID(uInt64ToSpanID(span.ParentID))
 			newSpan.SetName(span.Name)
 			newSpan.Status().SetCode(ptrace.StatusCodeOk)
+			newSpan.Attributes().PutStr("span.Resource", span.Resource)
```
Comment (hughsimpson) [inline receiver/datadogreceiver/translator.go:87]: I guess the name chosen here is probably the thing worth bikeshedding over? Could call it `dd.span.Resource` or something... 🤷

Comment (jpkrohling) [reply receiver/datadogreceiver/translator.go:87]: I agree. People can use the transform processor if they want another name for this.

**Comment by hughsimpson** [comment]
@boostchicken could you check this over? Seems a pretty simple change to me...

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**